Walking speed fix2

cstamas opened the pull request:
Two commits. One "promotes" only to core the Jackson based marshallers, is totally harmless.

Second commit is the key, it makes JacksonJSON Marshaller used by LS AttributeStorage.

cstamas committed:
Promoting Jackson based Marshallers to Core.

cstamas committed:
Switching to Jackson JSOM Marshaller.

mcculls commented on nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/attributes/DefaultLSAttributeStorage.java:
```diff
@@ -55,7 +55,7 @@
      */
     @Inject
-    public DefaultLSAttributeStorage( @Named( "xstream-xml" ) final Marshaller marshaller )
+    public DefaultLSAttributeStorage( @Named( "jackson-json" ) final Marshaller marshaller )
```
Does this change the storage format from XML to JSON? What about existing marshalled attribute files in XML format?

cstamas replied:
Yes, it does.

Since Nexus 2.0 is not yet released, users would be not affected at all, since upgrade would happen from legacy attribute storage (using XStream legacy XML format) to new attribute storage (using this Jackson JSON format).

Only one single problem is RSO, where attributes are upgraded already (using M4).

velo replied:
I'm happy with this, but are there any tests covering this?

bdemers commented:
we should leave JacksonXMLMarshaller in test until we need to move it

velo commented:
+1 after CI

adreghiciu commented:
+1, looking at the speed improvements

cstamas committed:
JacksonXML Marshaller moved back to test.

Also, the JacksonXML Mapper moved along, as it's not needed
and should not be pinned to core (keep core deps minimal)

cstamas commented:
JacksonXML Marshaller (and corresponding deps) moved back to test to not pollute core deps (and keep 'em on minimum)

cstamas commented:
CI job
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/195/

mcculls commented:
+1

nabcos commented:
+1

cstamas committed:
This IT is Tampering with attributes on new location directly.

Hence, is affected by storage format change (Xstream > JacksonJSON)
